src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http2.cs

```diff
@@ -1990,6 +1991,56 @@ public async Task SendAsync_ConcurentSendReceive_Fail()
             });
         }
 
+        [Fact]
+        [OuterLoop("Waits for seconds for events that shouldn't happen")]
+        public async Task SendAsync_StreamContentRequestBody_WaitsForRequestBodyToComplete()
+        {
+            var waitToSendRequestBody = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
+            var sendAsyncCompleted = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
+
+            // Create a content stream that will wait for a signal before it dribbles out some request content.
+            int sent = 0;
+            var stream = new DelegateStream(
+                canReadFunc: () => true,
+                readAsyncFunc: async (buffer, offset, count, token) =>
+                {
+                    await waitToSendRequestBody.Task;
+                    return sent++ < 10 ? 1 : 0;
+                });
+
+            await Http2LoopbackServer.CreateClientAndServerAsync(async url =>
+            {
+                using (HttpClient client = CreateHttpClient())
+                {
+                    // Connect to the server and SendAsync.
+                    var request = new HttpRequestMessage(HttpMethod.Post, url);
+                    request.Version = new Version(2, 0);
+                    request.Content = new StreamContent(stream);
+                    Task<HttpResponseMessage> sendAsyncTask = client.SendAsync(request, HttpCompletionOption.ResponseHeadersRead);
+
+                    // Wait a bit, and confirm SendAsync hasn't completed yet (because the request content can't have finished yet).
+                    await Task.Delay(1000);
+                    Assert.False(sendAsyncTask.IsCompleted);
+
+                    // Now let the request content go.  The SendAsync task should complete quickly.
+                    waitToSendRequestBody.SetResult(true);
+                    using (HttpResponseMessage r = await sendAsyncTask)
+                    {
+                        // Wake up the
```

Comment:
"Wake up the" -- missing some text in the comment?

Reply:
Oops.  "...server".